lua/telescope/previewers.lua

```diff
@@ -34,6 +39,84 @@ end
 
 local add_quotes = valuate_shell()
 
+local all_fts = vim.tbl_filter(function(line)
+  return line:find('setf')
+end, vim.fn.split(vim.fn.execute('autocmd filetypedetect'), '\n'))
+
+local ft_cache = {
+  Makefile = 'make', makefile = 'make', c = 'c', h = 'c', cpp = 'cpp', hpp = 'cpp',
+  css = 'css', ['CMakeLists.txt'] = 'cmake', Dockerfile = 'dockerfile', go = 'go',
+  js = 'javascript', lua = 'lua', py = 'python', vim = 'vim'
+}
+local determine_filetype = function(filepath)
+  local ext = vim.fn.fnamemodify(filepath, ':e')
+  if ext == '' then
+    local match = ft_cache[vim.fn.fnamemodify(filepath, ':t')]
+    if match then return match else return '' end
+  end
+
+  if ext == 'txt' then
+    local match = ft_cache[vim.fn.fnamemodify(filepath, ':t')]
+    if match then return match end
+    -- Take a look at end of file for vim:noet:tw=78:ts=8:ft=help:norl:
+    local tail = path.read_last_line(filepath)
+    if tail:sub(1, 5) == ' vim:' then
+      return tail:match('.*:ft=([^:]*):.*$')
+    end
+  end
```

Comment:
This part I would love in plenary. `plenary.filetype` and we can call `plenary.filetype.detect()` This would be useful for a lot of other plugins.

Reply:
Will work on this sometime in the future :)